src/cpu_feature/auxv.rs

```diff
@@ -0,0 +1,159 @@
+extern crate libc;
+extern crate byteorder;
+
+use std;
+use std::collections::HashMap;
+use std::vec::Vec;
+use std::io::BufRead;
+use self::byteorder::{NativeEndian, ReadBytesExt};
+use self::libc::c_ulong;
+
+// from [linux]/include/uapi/linux/auxvec.h. First 32 bits of HWCAP bits.
+pub const AT_HWCAP: c_ulong = 16;
+// currently only used by powerpc and arm64 AFAICT
+pub const AT_HWCAP2: c_ulong = 26;
+
+pub type AuxVals = HashMap<c_ulong, c_ulong>;
+
+#[derive(Debug, PartialEq)]
+pub enum AuxValError {
+    IoError,
+    InvalidFormat
+}
+
+/// Read an entry from the aux vector.
+///
+/// input: pairs of unsigned longs, as in /proc/self/auxv. The first of each
+/// pair is the 'type' and the second is the 'value'.
+///
+/// aux_types: the types to look for
+/// returns a map of types to values, only including entries for types that were
+/// requested that also had values in the aux vector
+#[allow(dead_code)] // TODO
+pub fn search_auxv(input: &mut BufRead, aux_types: &[c_ulong]) ->
+Result<AuxVals, AuxValError> {
+    let ulong_size = std::mem::size_of::<c_ulong>();
+    let mut buf: Vec<u8> = Vec::with_capacity(2 * ulong_size);
+    let mut result = HashMap::<c_ulong, c_ulong>::new();
+
+    loop {
+        buf.clear();
+        // fill vec so we can slice into it
+        for _ in 0 .. 2 * ulong_size {
+            buf.push(0);
+        }
+
+        let mut read_bytes = 0;
+        while read_bytes < 2 * ulong_size {
+            // read exactly buf's len of bytes.
+            match input.read(&mut buf[read_bytes..]) {
+                Ok(n) => {
+                    if n == 0 {
+                        // should not hit EOF before AT_NULL
+                        return Err(AuxValError::InvalidFormat)
+                    }
+
+                    read_bytes += n;
+                }
+                Err(_) => return Err(AuxValError::IoError)
+            }
+        }
+
+        let mut reader = &buf[..];
+        // TODO determine length to read
+        let found_aux_type = reader.read_u64::<NativeEndian>()
+            .map_err(|_| AuxValError::InvalidFormat)?;
+        let aux_val = reader.read_u64::<NativeEndian>()
+            .map_err(|_| AuxValError::InvalidFormat)?;
```

Comment:
Depends how much it is problematic to depend on byteorder... :) 

At first glance, https://github.com/BurntSushi/byteorder/blob/master/src/lib.rs is not enormous, so we could strip that down to only support the 32 bit and 64 bit cases, and then do some manual buffering rather than use their `Read` extension. Several hundred lines in total, assuming we maintain the tests. It certainly can be done, but if we keep anything like the same API it will mean keeping most of that library as an internal copy. 

If avoiding this particular dependency is of non-trivial importance, I can trace through the byteorder code and see if there's a more svelte way to extract what we need that would be more svelte.